options/posix/include/mlibc/lookup.hpp

```diff
@@ -44,8 +46,10 @@ struct ai_buf {
 
 int lookup_name_dns(struct lookup_result &buf, const char *name,
 		frg::string<MemoryAllocator> &canon_name);
+int lookup_addr_dns(frg::span<char> name, frg::array<uint8_t, 16> &addr, int family);
```

Comment:
`string_view` probably makes more sense than `span<char>`.

Reply:
A ``string_view`` is immutable, but I need to be able to overwrite the buffer

Reply:
Alright. So the result is fixed-length? If it isn't, it might make more sense to return a `string`.

Reply:
Yeah, it's always limited by the size of the buffer the user passes.